cmd/earthly/root_cmds.go

```diff
@@ -754,3 +805,31 @@ func (app *earthlyApp) actionPrune(cliCtx *cli.Context) error {
 	app.console.Printf("Freed %s\n", humanize.Bytes(total))
 	return nil
 }
+
+func handleDockerFiles(buildContextPath string) error {
+	earthfilePath := filepath.Join(buildContextPath, "Earthfile")
+	earthfilePathExists, err := fileutil.FileExists(earthfilePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to check if %q exists", earthfilePath)
+	}
+	if earthfilePathExists {
+		return errors.Errorf("earthfile already exists; please delete it if you wish to continue")
```

Comment:
does this mean it's impossible for a user to run `earthly docker-build ...` twice in a row (without having to delete the generated Earthfile)?

This would irritate me as a user, since it would slow down my dev workflow.

Reply:
since the file is removed in the end it shouldn't be a problem. This check is to make sure the user doesn't accidentally override an existing Earthfile. 
Switching to a temp dir can avoid this check though.